src/jshint.js

```diff
@@ -3518,6 +3303,30 @@ var JSHINT = (function() {
         }
       }
 
+      for (var t in tokens) {
+        if (tokens.hasOwnProperty(t)) {
+          t = tokens[t];
+          if (funct["(global)"]) {
+            if (predefined[t.id] === false) {
+              warning("W079", t.token, t.id);
```

Comment:
thanks fixed. I didn't actually add this code, just moved it within the function. While testing I found #2438 which is present before and after this PR is applied.